src/triage/component/catwalk/utils.py

```diff
@@ -204,13 +206,20 @@ def save_db_objects(db_engine, db_objects):
 
     Args:
         db_engine (sqlalchemy.engine)
-        db_objects (list) SQLAlchemy model objects, corresponding to a valid table
+        db_objects (iterator) SQLAlchemy model objects, corresponding to a valid table
```

Comment:
it would be nicer, anyway, not to insist that you receive an iterator, rather an "iterable," (so it can be a list, or a generator, _etc._).

then the first line can be anything like:

    db_objects = iter(db_objects)

to get your iterator.